src/Raven.Server/Utils/BackupUtils.cs

```diff
@@ -35,6 +35,59 @@ namespace Raven.Server.Utils;
 
 internal static class BackupUtils
 {
+    internal static BackupTask GetBackupTask(DocumentDatabase database, BackupParameters backupParameters, BackupConfiguration configuration, Logger logger, PeriodicBackupRunner.TestingStuff forTestingPurposes = null)
+    {
+        var destination = GetDestination();
+
+        switch (destination)
+        {
+            case DirectUploadBackupTask.SupportedDirectUploadDestination.S3:
+                return new DirectUploadBackupTask(destination.Value, database, backupParameters, configuration, logger, forTestingPurposes);
+
+            case null:
```

Comment:
I don't really like that `null` is a valid value and we have logic based on that. Should we have something like `IsDirectUploadSupported()` ?

Reply:
done